### PR TITLE
views: addition of custom error handlers 401, 403, 404

### LINKED ISSUE #13: views: add custom error handlers 401, 403, 404

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Added custom error handlers to errors 401,
  403 and 404. (Closes #13)

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen I have created the tests overriding the variable ```THEME_ERROR_TEMPLATE``` in order to avoid the assets fight. I am also creating a "fake.html" in a temp dir. I don't think this is the best way.

File: tests/conftest.py [hunk 1]
```diff
@@ -40,3 +46,31 @@ def app():
...
+
+@pytest.fixture()
+def app_error_handler():
+    """Flask app error handler fixture."""
+    app = Flask('myapp')
+
+    # Creation of a fake theme error template file.
+    temp_dir = tempfile.mkdtemp()
+    invenio_theme_dir = os.path.join(temp_dir, 'invenio_theme')
+    os.mkdir(invenio_theme_dir)
```
Comment (jirikuncar) [inline tests/conftest.py]: you should also clean-up (see https://pytest.org/latest/fixture.html#fixture-finalization-executing-teardown-code).

File: invenio_theme/views.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+"""Invenio user management and authentication."""
+
+from __future__ import absolute_import, print_function
+
+from flask import render_template
+
+
+def unauthorized(e):
+    """Error handler to show a 401.html page in case of a 401 error."""
+    return render_template('invenio_theme/401.html'), 401
```
Comment (lnielsen) [inline invenio_theme/views.py]: ``return render_template(current_app.config['THEME_404_TEMPLATE']), 401``

**Comment by jirikuncar** [comment]
`s/Added/Adds/`

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Corrected.

**Comment by jirikuncar** [comment]
@JavierDelgadoFernandez PR needs to be rebased.

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Done.

File: setup.py [hunk 3]
```diff
@@ -64,6 +65,7 @@
     "Flask-BabelEx>=0.9.2",
     "Flask-Breadcrumbs>=0.3.0",
     "Flask-Menu>=0.5.0",
+    "invenio-assets>=1.0.0a1",
```
Comment (lnielsen) [inline setup.py]: Is there a reason why this dependency is needed?

File: invenio_theme/views.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Invenio user management and authentication."""
```
Comment (lnielsen) [inline invenio_theme/views.py]: Seems like a copy/paste :-)

File: setup.py (hunk 3 above)
Comment (JavierDelgadoFernandez) [inline setup.py]: @lnielsen It is used in tests and in the example application. Should I remove it from the "install_requires"?

Comment (jirikuncar) [inline setup.py]: Yes. It should be only in tests_require.

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen Done.

**Pull request merged**